### DISCUSSION TITLE
How to get an inner port index in a Circuit global s-parameters

### DISCUSSION THREAD

**Discussion by tzortea** [scikit-rf/scikit-rf#1063] [Q&A] [answered]
If I want to view the s-parameters from an inner or external port to another inner or external port, how do I get the port indices, and is this the proper use of the global s-parameters?  Are the inner ports simply the connection nodes that are not connected to external ports?

**Comment by jhillairet** [comment, marked as answer]
The inner connexions of a `Circuit` is given by the `Circuit.connections_list` property. It lists the index associated to each (Network, port). 

For example, let's build the following circuit:
![image](https://github.com/scikit-rf/scikit-rf/assets/4642848/2bace416-a051-4a2c-b5ed-da3fbf9b249d)

```python
import skrf as rf

freq = rf.Frequency(1, 100, npoints=101, unit='MHz')

L0 = 1
R = 50
R1 = 25
C = 1e-10

med = rf.DefinedGammaZ0(frequency=freq, gamma=1j*freq.w/rf.c)

line1 = med.line(d=L0, unit='m', name='line1')
line2 = med.line(d=L0, unit='m', name='line2')

load= med.resistor(R=R, frequency=freq, name='load')
res = med.resistor(R=R1, frequency=freq, name='R')
cap = med.capacitor(C=C, frequency=freq, name='C')

port = rf.Circuit.Port(frequency=freq, name='port')
gnd = rf.Circuit.Ground(frequency=freq, name='gnd')

cnx = [
       [(port, 0), (line1, 0)],
       [(line1, 1), (res, 0)],
       [(res,1), (cap, 0), (line2, 0)],
       [(line2, 1), (load, 0)],
       [(cap, 1), (load, 1), (gnd, 0)]
       ]
cir = rf.Circuit(cnx)
ntw = cir.network
```

If you look to the `cir.connections_list`, you will have:
```python
[[0, (1-Port Network: 'port',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j], 0)],
 [1,
  (2-Port Network: 'line1',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)],
 [2,
  (2-Port Network: 'line1',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 1)],
 [3, (2-Port Network: 'R',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)],
 [4, (2-Port Network: 'R',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 1)],
 [5, (2-Port Network: 'C',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)],
 [6,
  (2-Port Network: 'line2',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)],
 [7,
  (2-Port Network: 'line2',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 1)],
 [8,
  (2-Port Network: 'load',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)],
 [9, (2-Port Network: 'C',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 1)],
 [10,
  (2-Port Network: 'load',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 1)],
 [11,
  (2-Port Network: 'gnd',  1.0-100.0 MHz, 101 pts, z0=[50.+0.j 50.+0.j], 0)]]
```
The numbers in front of each tuple (Network, port) are the index in blue in the following figure:
![image](https://github.com/scikit-rf/scikit-rf/assets/4642848/2746efa2-ec44-4ffd-b3fc-04c7d5b4dea0)

There is 12 elements: so, the inner S-parameters of the circuit (the `cir.s` property) is indeed a nb_freqx12x12 matrix. 

On this example, since there is a single port defined, the external S matrix (`cir.s_external` or the resulting network from `cir.network`) will be a nb_freqx1x1 matrix.
